### PR TITLE
cura: support plugins, add octoprint-plugin

### PR DESCRIPTION
**Pull request by betaboon**
###### Motivation for this change
I wanted to be able to install plugins.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/cura/plugins.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      nativeBuildInputs = [ cmake ];
+
+      propagatedBuildInputs = with python3Packages; [
+        netifaces
+      ];
+
+      meta = with stdenv.lib; {
+        description = "Enables printing directly to OctoPrint and monitoring the process";
+        homepage = https://github.com/fieldOfView/Cura-OctoPrintPlugin;
+        license = licenses.agpl3;
```
Comment (gebner) [inline pkgs/applications/misc/cura/plugins.nix]: Please add yourself as a maintainer here.

File: pkgs/applications/misc/cura/plugins.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      nativeBuildInputs = [ cmake ];
+
+      propagatedBuildInputs = with python3Packages; [
+        netifaces
+      ];
+
+      meta = with stdenv.lib; {
+        description = "Enables printing directly to OctoPrint and monitoring the process";
+        homepage = https://github.com/fieldOfView/Cura-OctoPrintPlugin;
```
Comment (gebner) [inline pkgs/applications/misc/cura/plugins.nix]:
```suggestion
        homepage = "https://github.com/fieldOfView/Cura-OctoPrintPlugin";
```

URLs should be quoted now: https://github.com/NixOS/rfcs/pull/45

File: pkgs/applications/misc/cura/plugins.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  self = {
+
+    octoprint = stdenv.mkDerivation rec {
+      name = "Cura-OctoPrintPlugin";
+      version = "3.5";
+
+      src = fetchFromGitHub {
+        owner = "fieldOfView";
+        repo = "Cura-OctoPrintPlugin";
+        rev = version;
```
Comment (gebner) [inline pkgs/applications/misc/cura/plugins.nix]: In the `Cura-OctoPrintPlugin` repository, `3.5` is a branch and not a tag.  That is, it changes regularly and does not point to a fixed commit.  Please just put the latest revision here.

**Comment by gebner** [comment]
Sorry, I completely missed this.  I tested it with `cura.override { plugins = [ curaPlugins.octoprint ]; }`, and I can successfully connect to my octoprint server.

Unfortunately there doesn't seem to be any easier way to register the plugins since cura looks for them in the installation directory.  But fortunately building cura is cheap, it's just copying python files.

**Comment by betaboon** [comment]
@gebner thanks for taking care of this. i have been away for a while and haven't found the time to take care of your notes. sorry for that.

**Pull request merged**